SwiftUI.md

```diff
@@ -0,0 +1,61 @@
+# SwiftUI
+
+## GeometryReader
+
+### Convention
+Avoid using [`GeometryReaders`](https://developer.apple.com/documentation/swiftui/geometryreader) in your main view tree. To measure views, place `GeometryReaders` inside of a `.background` or `.overlay` and communicate the size up through a `Preference` or other methods.
+
+### Rationale
+`GeometryReaders` always fill all of their proposed size, essentially overriding the layout behavior of the views they wrap. They also override the children's ideal size.
+
+Usage of `.background` or `.overlay` is recommended because the content of these modifiers doesn't affect the layout of the parent view, so this limits the GR's impact while still allowing you to measure the parent view final size (note this can be different from the proposed size).
+
+### Example
+```swift
+// Good: Measure final size of Text without affecting its layout
+Text("Sample text")
+    .background { // Size proposed to this will be the final size of the Text view
+        GeometryReader { proxy in
+            Color.clear.preference(key: SizePreferenceKey.self, value: proxy.size)
+        }
+    }
+    .onPreferenceChange(SizePreferenceKey.self) { size in
+        // Do things with the measured size
+    }
+
+private struct SizePreferenceKey: PreferenceKey {
+    static var defaultValue: CGSize = .zero
+
+    static func reduce(value: inout CGSize, nextValue: () -> CGSize) {
+        value = nextValue()
+    }
+}
+```
+
+```swift
+// Bad: We measure size proposed to the Text view, but the GeometryReader won't fit the text. 
+GeometryReader { proxy in 
+    Text("Sample text")
+}
+```
+
+### Exceptions
+There are cases where you may need to read the size proposed to a view, instead of the final size (for example to do custom proportional layout). In these cases, first consider if a combination of stacks, spacers and frames can be used instead.
+
+If you absolutely need the `GeometryReader`, expanding to fill the dimension you want to measure is unavoidable. You can still limit the impact using a few of these methods:
+- Wrap the `GeometryReader` in a frame, constraining the dimension you don't care about.
+- Place the geometry reader in a `ZStack`. Combined with the previous method, this can help you measure without affecting the layout of other views in the `ZStack`. Note this will still cause the `ZStack` to fill the dimension you're measuring.
+
+## Multiline `UILabel` representables
```

Comment:
No code example for this?

Reply:
Thoughts on the expanded GeometryReader section?

Reply:
It kind of diverged from a pure "convention - example" format into a more detailed guide, but I think it's good content.